### PR TITLE
Added czmq package, with example

### PR DESCRIPTION
**Pull request by ErniBrown**
Added czmq package, as requested by @roalz in #599 

Using [hunter-packages/czmq](https://github.com/hunter-packages/czmq), which add cmake config files and use standard cmake scheme

### CODE REVIEW

File: examples/czmq/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,16 @@
+cmake_minimum_required (VERSION 3.2)
+
+# Emulate HunterGate:
+# * https://github.com/hunter-packages/gate
+include("${CMAKE_CURRENT_SOURCE_DIR}/../common.cmake")
+
+project(download-czmq)
+
+set(THREADS_PREFER_PTHREAD_FLAG ON)
```
Comment (ruslo) [inline examples/czmq/CMakeLists.txt]: There is no `pthread` in user's code so it looks suspicious. Have you experienced any problems without this line?

File: examples/czmq/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+hunter_add_package(ZeroMQ)
```
Comment (ruslo) [inline examples/czmq/CMakeLists.txt]: If `ZeroMQ` is the dependency of `czmq` then we don't need to add it explicitly. Hunter will manage it automatically.

File: cmake/projects/czmq/hunter.cmake [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    "3.0.0-p0"
+    URL
+    "https://github.com/hunter-packages/czmq/archive/v3.0.0-p0.tar.gz"
+    SHA1
+    b5dce40c8239ad1d6dbf87fb96ac3123f1c392d8
+)
+
+hunter_pick_scheme(DEFAULT url_sha1_cmake)
+hunter_cacheable(czmq)
+hunter_download(PACKAGE_NAME czmq)
```
Comment (ruslo) [inline cmake/projects/czmq/hunter.cmake]: http://0.readthedocs.io/en/latest/git.html#check-there-is-a-new-line-at-end-of-file

Comment (ErniBrown) [inline cmake/projects/czmq/hunter.cmake]: This is embarassing...

File: examples/czmq/CMakeLists.txt (hunk 2 above)
Comment (ErniBrown) [inline examples/czmq/CMakeLists.txt]: No, you are right, this and the next comment are both remains of a previous code. I will remove the lines and squash everything

**Comment by ErniBrown** [comment]
Updated!

**Comment by ruslo** [comment]
Testing:
* https://ci.appveyor.com/project/ingenue/hunter/build/1.0.1208
* https://travis-ci.org/ingenue/hunter/builds/180601867

**Comment by ruslo** [comment]
> https://travis-ci.org/ingenue/hunter/builds/180601867

Waiting for OSX machines, problems on Travis side:
* https://www.traviscistatus.com/incidents/nq0sf0srvx8s

**Comment by ruslo** [comment]
> Waiting for OSX machines

Finished. Some tests failed, @ErniBrown please send pull request with excluding broken toolchains to branch [pkg.czmq](https://github.com/ingenue/hunter/tree/pkg.czmq) or do the fixes.

**Comment by ErniBrown** [comment]
[Build 2282](https://travis-ci.org/ruslo/hunter/builds/180372211) failed because of apt-get errors, probably related to the virtual machine infrastructure. Anyway it was outdated by [build 2283](https://travis-ci.org/ruslo/hunter/builds/180394346), which did not failed, but was stopped.
I think the test must be rerun, should I trigger it somehow?

**Comment by ruslo** [comment]
> Build 2282 failed because of apt-get errors, probably related to the virtual machine infrastructure. Anyway it was outdated by build 2283, which did not failed, but was stopped

You're looking at the wrong links. This one are the unit tests and I stop them because your pull request doesn't change internal stuff, hence unit tests are not very interesting. Take a look at the links in this comment: https://github.com/ruslo/hunter/pull/603#issuecomment-264364141

**Comment by ErniBrown** [comment]
I Ruslo,
I'm finally back on my project using hunter. I would be interested into integrating my hunter fork in the main repo, so I was asking if is it possible to reopen the pull request, or if I can start a new one.
I need some information, I updated the PR to the last develop commit, but i think I need to provide tests for continuous integration, right? Where should I look into for this?
 https://github.com/ingenue/hunter/tree/pkg.czmq?
Thanks,
E

**Comment by ruslo** [comment]
>  to reopen the pull request, or if I can start a new one

Please start the new one.

> but i think I need to provide tests for continuous integration, right?

Not a test, just an example which will test hunter_add_package/find_package/target_link_libraries chain. See for example GTest:
* https://github.com/ruslo/hunter/blob/master/examples/GTest/CMakeLists.txt

**Pull request closed without merging**